### PR TITLE
asset upload

### PR DESCRIPTION
**Pull request by officialpranav**

### CODE REVIEW

File: lib/admin/consts.dart [hunk 1]
```diff
@@ -25,6 +25,27 @@ const workType = {
   'SAF': 'Safety',
 };
 
+const assetWorkType = {
+  "BDE": {"Description" : "Breakdown - Electrical", "Priority" : "4"},
+  "BDM": {"Description" : "Breakdown - Mechanical", "Priority" : "4"},
```
Comment (jonathanmajh) [inline lib/admin/consts.dart:30]: Should be a map of objects instead of map of maps

File: lib/admin/db_drift.dart [hunk 2]
```diff
@@ -201,18 +204,65 @@ class MyDatabase extends _$MyDatabase {
...
+  ///
+  ///1 = new asset
+  ///
+  ///-1 = asset that failed upload (e.g. location and asset were uploaded, but job plan failed)
+  Future<int> setAssetStatus(
+      String assetNum, String siteId, int assetStatus) async {
+    var existingAsset = await getAsset(siteId, assetNum);
+    var newAsset = Asset(
+        assetnum: existingAsset.assetnum,
+        changedate: existingAsset.changedate,
```
Comment (jonathanmajh) [inline lib/admin/db_drift.dart:246]: seems extra, see first example
https://drift.simonbinder.eu/docs/getting-started/writing_queries/#updates-and-deletes

File: lib/creation/asset.dart [hunk 3]
```diff
@@ -0,0 +1,818 @@
...
+        ),
+      ),
+      endDrawer: const EndDrawer(),
+      body: TabBarView(
+        controller: _tabController,
+        children: [
+          Column(
+            children: <Widget>[
+              ElevatedButton(
+                child: Text('print pending'),
```
Comment (jonathanmajh) [inline lib/creation/asset.dart:107]: remove

File: lib/creation/asset.dart [hunk 4]
```diff
@@ -0,0 +1,818 @@
...
+                      .pendingAssets
+                      .entries
+                      .toList()) {
+                    print("${entry}\n\n");
+                  }
+                },
+              ),
+              ElevatedButton(
+                child: const Text('print assets'),
+                onPressed: () async {
```
Comment (jonathanmajh) [inline lib/creation/asset.dart:121]: remove

File: lib/admin/upload_maximo.dart [hunk 5]
```diff
@@ -302,6 +307,318 @@ Future<Map<String, List<List<String>>>> uploadToMaximo(
...
 
+///Uploads assets to maximo.
+///
+///Order of upload: Location > Asset > JobPlan > JobLabor > JobAsset
+///
+///@returns Map<String, dynamic> with keys 'location', 'asset', 'jobplan', 'joblabor', 'jpassetlink'
+///
+///Possible statuses for each key: 'success', 'fail', 'duplicate'.
+///In case of duplicate, the upload will continue
+Future<Map<String, dynamic>> uploadAssetToMaximo(
```
Comment (jonathanmajh) [inline lib/admin/upload_maximo.dart:318]: on second thought i would like to see one function for generating the upload, another one for actually doing the upload

File: lib/creation/pair.dart [hunk 6]
```diff
@@ -0,0 +1,108 @@
+//@author Pranav Bommireddipalli (iko's #1 employee 💯💯💯)
+///A class that stores two objects of different types
+///
+///Declaration:
```
Comment (jonathanmajh) [inline lib/creation/pair.dart:4]: i dont know what the point of this is

**Comment by officialpranav** [comment]
Comments resolved

**Comment by jonathanmajh** [comment]
Header should follow layout of other modules.
Title does show for light mode
image

image

image

**Comment by jonathanmajh** [comment]
Add asset number validation
1 Letter then 4 numbers
image

**Comment by jonathanmajh** [comment]
Batch add assets using clipboard (N)

**Comment by jonathanmajh** [comment]
Custom Standard job plan description (N)
current asset description length limit should be used
Asset description can be 100 instead

**Comment by jonathanmajh** [comment]
Put upload details to clipboard similar to PM module (N)

**Pull request merged**